### ISSUE TITLE
`motion_from_essential_choose_solution` silently returns candidate 0 when no point passes the depth test

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4879]
`motion_from_essential_choose_solution` triangulates the correspondences for each of the four `(R, t)` candidates and returns the one with the most points at positive depth in both cameras (`torch.max(depth_mask.sum(-1), ...)`, `essential.py:756`). When no point passes — for example when `mask` excludes every correspondence — all four counts are 0, `torch.max` returns index 0, and the function returns candidate 0 and its triangulated points with nothing to tell the caller. In the repro, with `mask` all `False` it returns a rotation 2.0 (max-abs entry difference) away from the true one and points at depths down to −80.9, where the same call without the mask recovers the pose to 6e-8. OpenCV's `recoverPose` returns the number of points that passed the check, so a caller can detect 0; kornia's `(R, t, points3d)` carries no count.

## Repro

```python
import torch
from kornia.geometry.conversions import axis_angle_to_rotation_matrix
from kornia.geometry.epipolar import essential_from_Rt, motion_from_essential, motion_from_essential_choose_solution

K1 = torch.tensor([[[800.0, 0, 320], [0, 760, 200], [0, 0, 1]]])
K2 = torch.tensor([[[700.0, 0, 300], [0, 740, 240], [0, 0, 1]]])
R = axis_angle_to_rotation_matrix(torch.tensor([[0.1, -0.2, 0.05]]))
t = torch.tensor([[[0.5], [0.1], [0.02]]])
torch.manual_seed(0)
X = torch.rand(12, 3) * 2 - 1
X[:, 2] += 5
x1 = (K1[0] @ X.T).T
x2 = (K2[0] @ (R[0] @ X.T + t[0])).T
x1, x2 = (x1[:, :2] / x1[:, 2:])[None], (x2[:, :2] / x2[:, 2:])[None]
E = essential_from_Rt(torch.eye(3)[None], torch.zeros(1, 3, 1), R, t)
Rs, ts = motion_from_essential(E)

Rok, tok, _ = motion_from_essential_choose_solution(E, K1, K2, x1, x2)
print("all points usable:  |R - R_true| =", f"{(Rok - R).abs().max().item():.2g}", " t =", tok.flatten().tolist())

# every correspondence masked out: no candidate has a single point in front of both cameras
none = torch.zeros(1, 12, dtype=torch.bool)
Rn, tn, Xn = motion_from_essential_choose_solution(E, K1, K2, x1, x2, mask=none)
print("mask all False:     |R - R_true| =", f"{(Rn - R).abs().max().item():.2g}", " t =", tn.flatten().tolist())
print("   returned pose is candidate 0:", torch.equal(Rn, Rs[:, 0]) and torch.equal(tn, ts[:, 0]),
      "| points3d finite:", torch.isfinite(Xn).all().item(), "| min depth:", f"{Xn[..., 2].min().item():.3g}")
```

Output:

```text
all points usable:  |R - R_true| = 6e-08  t = [0.9798272252082825, 0.1959654539823532, 0.039193086326122284]
mask all False:     |R - R_true| = 2  t = [0.9798272252082825, 0.1959654539823532, 0.039193086326122284]
   returned pose is candidate 0: True | points3d finite: True | min depth: -80.9
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same), CPU float32, macOS arm64.

## Classification

API contract gap. Idiom 4 (degenerate or singular input silently producing garbage instead of raising). Low priority: a caller reaches it only when every correspondence fails or is masked out.

## Disposition

MERGE IN WINDOW — behaviour change: return a validity signal with the pose (the per-element count of points that passed the depth test, as `recoverPose` does, or a boolean flag), in the shape the validity-mask discussion in #4555 settles on; the return tuple changes.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Related: #2198 (same function, batched index).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
